tests/lang/ArrayTest.php

```diff
@@ -398,4 +398,91 @@ public function testGetWithNotExistentIndex(): void {
 		$element = $fruits->get(4);
 		$this->assertNull($element);
 	}
+
+	public function testMerge(): void {
+		$fruits = new ArrayObject(['apple', 'apricot', 'peach', 'banana']);
+		$fruits->merge(['ananas', 'watermelon']);
+		$expected = ['apple', 'apricot', 'peach', 'banana', 'ananas', 'watermelon'];
+		$this->assertEquals($expected, $fruits->toArray());
+	}
+
+	public function testMergeMultidimensional(): void {
+		$cartoons = new ArrayObject([
+			'Marvel' => [
+				'avengers' => [
+					'Iron Man',
+					'Hulk',
+					'Captain America'
+				]
+			],
+			'Dc' => [
+				'Superman',
+				'Bat Man'
+			]
+		]);
+
+		$goNagai = [
+			'Go Nagai' => [
+				'Mazinger-Z',
+				'Great Mazinger',
+				'Goldrake'
+			]
+		];
+
+		$addMarvel = [
+			'Marvel' => [
+				'X-Men' => [
+					'Wolverine',
+					'Professor X',
+					'Magneto'
+				]
+			]
+		];
+
+		$expected = [
+			'Marvel' => [
+				'X-Men' => [
+					'Wolverine',
+					'Professor X',
+					'Magneto'
+					]
+				],
+			'Dc' => [
+				'Superman',
+				'Bat Man'
+				],
+			'Go Nagai' => [
+				'Mazinger-Z',
+				'Great Mazinger',
+				'Goldrake'
+				]
+			];
+
+		$this->assertEquals($expected, $cartoons->merge($addMarvel, $goNagai)->toArray());
```

Comment:
I think the `$expected` then should not be handmade but also the result of `array_merge()` to stay in parity with the original function.

Reply:
Agree